lib/rules/fixer-return.js

```diff
@@ -56,20 +57,27 @@ module.exports = {
     }
 
     return {
+      Program (node) {
+        contextIdentifiers = utils.getContextIdentifiers(context, node);
+      },
 
       // Stacks this function's information.
       onCodePathStart (codePath, node) {
         const parent = node.parent;
+        const shouldCheck = node.type === 'FunctionExpression' &&
+          node.body.type === 'BlockStatement' &&
+          // check if it is context.report({fix})
+          utils.getKeyName(parent) === 'fix' &&
+          parent.parent.type === 'ObjectExpression' &&
```

Comment:
This will fail for the old-style `context.report` calls:

```js
context.report(node, 'foo', {}, fixer => {});
```

You can use [`utils.getReportInfo`](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/4cec353c272d6dd799ba3c30cd4f06b103b0c7e8/lib/utils.js#L192) to get info about a report.

I think the best way to do it would be to use something like this:

```diff
        const shouldCheck = node.type === 'FunctionExpression' &&
-         node.body.type === 'BlockStatement' &&
-         // check if it is context.report({fix})
-         utils.getKeyName(parent) === 'fix' &&
          parent.parent.type === 'ObjectExpression' &&
          parent.parent.parent.type === 'CallExpression' &&
          contextIdentifiers.has(parent.parent.parent.callee.object) &&
-         parent.parent.parent.callee.property.name === 'report';
+         parent.parent.parent.callee.property.name === 'report' &&
+         utils.getReportInfo(parent.parent.parent.arguments).fix === node;
```